### PR TITLE
update: Dockerfile, documentation

### PR DESCRIPTION
**Pull request by gyuho**
```
sudo docker pull quay.io/coreos/etcd:v3.0.0
sudo docker run --name etcd quay.io/coreos/etcd:v3.0.0 /usr/local/bin/etcd --version
sudo docker run --name etcd quay.io/coreos/etcd:v3.0.0 /bin/sh -c "/usr/local/bin/etcd --version"
sudo docker run --name etcd quay.io/coreos/etcd:v3.0.0 -e "ETCDCTL_API=3" /usr/local/bin/etcdctl version

sudo docker run --name etcd quay.io/coreos/etcd:v3.0.0 /usr/local/bin/etcd
sudo docker exec etcd /bin/sh -c "export ETCDCTL_API=3 && /usr/local/bin/etcdctl version"
```

I test-build docker image, now at https://quay.io/repository/coreos/etcd?tab=tags.

/cc @xiang90 @heyitsanthony

### CODE REVIEW

**Comment by xiang90** [comment]
what is the size different between the old image and this one?

File: Dockerfile [hunk 1]
```diff
@@ -1,2 +1,16 @@
-FROM golang:onbuild
+FROM alpine:latest
+RUN apk update
+RUN apk --update add bash
```
Comment (heyitsanthony) [inline Dockerfile]: does alpine have /bin/sh built-in? should probably use that if it's there

**Comment by gyuho** [comment]
@xiang90 `13.5MB` vs `8.1MB` (old)

File: scripts/build-docker [hunk 2]
```diff
@@ -9,11 +9,18 @@ mkdir -p ${IMAGEDIR}
 cp ${BINARYDIR}/etcd ${BINARYDIR}/etcdctl ${IMAGEDIR}
 
 cat <<DF > ${IMAGEDIR}/Dockerfile
```
Comment (heyitsanthony) [inline scripts/build-docker]: Any way to use ./Dockerfile here instead of copying it?

**Comment by gyuho** [comment]
@heyitsanthony Now it only uses `sh` and changed the script just to use `Dockerfile` from top directory.

Thanks.

File: scripts/build-docker [hunk 3]
```diff
@@ -8,12 +8,6 @@ IMAGEDIR=${BUILDDIR}/image-docker
 mkdir -p ${IMAGEDIR}
 cp ${BINARYDIR}/etcd ${BINARYDIR}/etcdctl ${IMAGEDIR}
 
-cat <<DF > ${IMAGEDIR}/Dockerfile
-FROM scratch
-ADD etcd /
-ADD etcdctl /
-EXPOSE 2379 2380
-ENTRYPOINT ["/etcd"]
```
Comment (heyitsanthony) [inline scripts/build-docker]: should ./Dockerfile keep `ENTRYPOINT`?

File: Dockerfile [hunk 4]
```diff
@@ -1,2 +1,12 @@
-FROM golang:onbuild
+FROM alpine:latest
+RUN apk update
```
Comment (heyitsanthony) [inline Dockerfile]: I don't think this is needed anymore?

Comment (gyuho) [inline Dockerfile]: yeah right. will fix.

**Comment by gyuho** [comment]
@heyitsanthony Addressed. Default entry point is now `etcd`. PTAL.

Thanks.

**Comment by heyitsanthony** [comment]
lgtm, defer to @xiang90

File: Documentation/op-guide/container.md [hunk 5]
```diff
@@ -43,12 +45,22 @@ sudo docker run --net=host --name etcd quay.io/coreos/etcd:${ETCD_VERSION} \
...
+	/usr/local/bin/etcd \
     --data-dir=data.etcd --name ${THIS_NAME} \
 	--initial-advertise-peer-urls http://${THIS_IP}:2380 --listen-peer-urls http://${THIS_IP}:2380 \
 	--advertise-client-urls http://${THIS_IP}:2379 --listen-client-urls http://${THIS_IP}:2379 \
 	--initial-cluster ${CLUSTER} \
 	--initial-cluster-state ${CLUSTER_STATE} --initial-cluster-token ${TOKEN}
 ```
 
-TODO: current etcd docker container is shipped with etcdctl, but the `ETCDCTL_API` environment value is not set inside the container. So currently there's no way to use etcdctl for v3 directly from the container (e.g. `docker exec etcd /etcdctl put foo bar` won't work).
+To run `etcdctl`:
```
Comment (xiang90) [inline Documentation/op-guide/container.md]: To run `etcdctl` using API version 3:

```
docker exec etcd /bin/sh -c "export ETCDCTL_API=3 && /usr/local/bin/etcdctl put foo bar"
```

**Comment by gyuho** [comment]
@xiang90 PTAL. Cleaned up the docker command.

**Comment by xiang90** [comment]
lgtm

**Pull request merged**